lib/Dialect/FIRRTL/Transforms/Dedup.cpp

```diff
@@ -154,16 +154,20 @@ struct StructuralHasher {
   }
 
 private:
-  void update(const void *pointer) {
-    auto *addr = reinterpret_cast<const uint8_t *>(&pointer);
-    sha.update(ArrayRef<uint8_t>(addr, sizeof pointer));
-  }
-
-  void update(size_t value) {
+  /// Hash value as array of bytes.
+  template <typename T>
```

Comment:
FWIW:
If I could confidently enable_if this to ensure it doesn't get used with things that have possibly unset padding or short reads or something I would but instead I just wrap it for types that are definitely safe and cover our needs.